flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/InternalOperatorCoordinatorMetricGroup.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.metrics.groups;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.metrics.MetricGroup;
+import org.apache.flink.metrics.groups.OperatorCoordinatorMetricGroup;
+
+/** Special {@link org.apache.flink.metrics.MetricGroup} representing an OperatorCoordinator. */
+@Internal
+public class InternalOperatorCoordinatorMetricGroup extends ProxyMetricGroup<MetricGroup>
+        implements OperatorCoordinatorMetricGroup {
+
+    public InternalOperatorCoordinatorMetricGroup(JobManagerOperatorMetricGroup parent) {
```

Comment:
```suggestion
    public InternalOperatorCoordinatorMetricGroup(MetricGroup parent) {
```
Don't hard-code the hierarchy into metric groups unless there's a good reason to do so.

Reply:
I think the InternalOperatorCoordinatorMetricGroup should be only registered under the  JobManagerOperatorMetricGroup. So I add this limit.

Reply:
That's not something the InternalOperatorCoordinatorMetricGroup has to concern itself with, and as-is it just makes testing more difficult.

Reply:
OK, I will apply this change to the other PR, which will introduce the internal operator coordinator metric group.